distributed/cmd/orchestration/main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"distributed/pkg/networking"
+	"distributed/pkg/networking/replication"
+	"distributed/pkg/server"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"sync"
+	"syscall"
+	"time"
+)
+
+const (
+	defaultDbDirectory   = "/tmp/distkv"
+	routingSocketPath    = "routing.sock"
+	orchestrationSocket  = "orchestration.sock"
+	replicaCommandSocket = "replica.sock"
+	primaryCommandSocket = "primary.sock"
+)
+
+var (
+	dbDirectory         string
+	launchedProcesses   []*os.Process
+	openedSockets       []net.Listener
+	orchestrationLogger *log.Logger
+)
+
+func main() {
+	flag.StringVar(
+		&dbDirectory,
+		"db",
+		defaultDbDirectory,
+		"directory for database files (primary and replicaSockets)",
+	)
+	flag.Parse()
+	orchestrationLogger = log.New(os.Stderr, "[orchestration] ", log.LstdFlags|log.Lmsgprefix)
+
+	fPath, err := filepath.Abs(filepath.Dir(os.Args[0]))
+	if err != nil {
+		orchestrationLogger.Fatalf("error getting filepath: %v", err)
+	}
+	serverExecutable := filepath.Join(fPath, "../server/server")
+
+	// teardown code
+	var (
+		endProgramWg sync.WaitGroup
+		sigChan      = make(chan os.Signal)
+	)
+	endProgramWg.Add(1)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+
+	go func(sigChan <-chan os.Signal, wg *sync.WaitGroup) {
+		<-sigChan
+		for _, process := range launchedProcesses {
+			if process != nil {
+				if err := process.Kill(); err != nil {
+					orchestrationLogger.Printf("error killing process %d: %v", process.Pid, err)
+				}
+			}
+		}
+		for _, socket := range openedSockets {
+			if err := os.Remove(socket.Addr().String()); err != nil {
+				orchestrationLogger.Printf("error closing socket %s: %v", socket, err)
+			}
+		}
+		wg.Done()
```

Comment:
Don't think I need this anymore now that I have `Router.Run()`.  had it installed when the orchestrator wasn't waiting for client connections.